### DISCUSSION TITLE
Are Nuxt layers cached by giget?

### DISCUSSION THREAD

**Discussion by saschafuchs** [nuxt/nuxt#28742] [Questions]
Had the strange problem today that one of the layers didn't want to update (in our case the layers are stored as git repo on github). We tried many things, such as deleting the node_module, .nuxt and dist folders. During the `npm install` the layers were downloaded and were always the old version, i.e. a version that was not available on the dev and main branches. After a few hours I deleted the repo locally and downloaded it again, since then it works again. 
Is there a flag that can be passed through so that the layers are no longer cached, or always clear the cache with each build?

**Comment by pi0** [comment]
Giget uses full url and response etag for cache invalidation (which is provided by github/gitlab at least).

The simplest way to force invalidate cache is adding a query param to the end of the layer URL like `?timestampt=${Date.now()}`.

You can see more debugging info by setting `DEBUG=1` environment variable and if needed more investigation, feel free to make an issue in unjs/giget also sharing an example URL input that is not being invalidated.

**Comment by saschafuchs** [comment]
Thank you for the information.
